### PR TITLE
Issue SB-4760 fix:Unsubscribe from observables(Search module)

### PR DESCRIPTION
**Pull request by shakkeel1330**
Issue SB-4760 fix:Unsubscribe from observables(Search module)

### CODE REVIEW

File: src/app/client/src/app/modules/search/components/user-edit/user-edit.component.ts [hunk 1]
```diff
@@ -241,5 +252,8 @@ export class UserEditComponent implements OnInit, OnDestroy {
   }
   ngOnDestroy() {
     this.modal.deny();
+    this.unsubscribe$.next();
+    this.unsubscribe$.complete();
+    this.permissionSubscription.unsubscribe();
```
Comment (anoophm) [inline src/app/client/src/app/modules/search/components/user-edit/user-edit.component.ts]: if(this.permissionSubscription) then unsub

File: src/app/client/src/app/modules/search/components/user-profile/user-profile.component.ts [hunk 2]
```diff
@@ -233,7 +244,9 @@ export class UserProfileComponent implements OnInit {
         }
       };
       this.userDetails.badgeArray = [];
-      this.badgesService.getDetailedBadgeAssertions(req, this.userDetails.badgeAssertions).subscribe((detailedAssertion) => {
+      this.badgesService.getDetailedBadgeAssertions(req, this.userDetails.badgeAssertions)
+      .takeUntil(this.unsubscribe$)
```
Comment (anoophm) [inline src/app/client/src/app/modules/search/components/user-profile/user-profile.component.ts]: follow rxjs 6 syntax

File: src/app/client/src/app/modules/search/components/user-edit/user-edit.component.ts [hunk 3]
```diff
@@ -172,7 +181,9 @@ export class UserEditComponent implements OnInit, OnDestroy {
         mainRole.push(value.role);
       });
       const option = { userId: this.userId, orgId: this.selectedOrgId, roles: roles };
-      this.userSearchService.updateRoles(option).subscribe(
+      this.userSearchService.updateRoles(option)
+      .takeUntil(this.unsubscribe$)
```
Comment (anoophm) [inline src/app/client/src/app/modules/search/components/user-edit/user-edit.component.ts]: follow rxjs 6 syntax, use pipe

**Comment by harishgilimi** [comment]
closing because pending from long

**Pull request closed without merging**